wled00/wled_server.cpp

```diff
@@ -15,6 +15,22 @@ bool isIp(String str) {
   return true;
 }
 
+void handleUpload(AsyncWebServerRequest *request, const String& filename, size_t index, uint8_t *data, size_t len, bool final){
+  if(!index){
+    request->_tempFile = WLED_FS.open(filename, "w");
```

Comment:
The PR suggests only five filenames to be allowed for upload.

If I am correct that any client could call this upload endpoint, then it appears that this opens the ability to upload *arbitrary* filenames (and contents) via this endpoint.

For example, is there anything to prevent uploading `/wsec.json` using this new upload interface? _[[using `curl`, not the WLED UI]]_

Should this code validate the filename matches one of a list of allowed filenames?

Reply:
Indeed it could be possible to upload _any_ file. Though internal files are never overwritten (and cannot be).
If you want to upload wsec.json you can, but this does not pose a security issue since you cannot read it.

Will think about a possible solution.

Reply:
I hadn't realized the internal files cannot be overwritten.  That's great, as it prevents changing the wifi settings page HTML (for example) to submit wifi credentials elsewhere.  Given that, uploading of any file is not really an issue.  👍